cpp/ql/lib/semmle/code/cpp/security/BufferAccess.qll

```diff
@@ -125,10 +129,7 @@ class StrncpyBA extends BufferAccess {
     result = this.(FunctionCall).getArgument(0) and
     bufferDesc = "destination buffer" and
     accessType = 2
-    or
-    result = this.(FunctionCall).getArgument(1) and
-    bufferDesc = "source buffer" and
-    accessType = 2
```

Comment:
@MathiasVP and I chatted a bit about this. This library is only used to support the overflow query if I recall correctly, so the easiest solution to correct the false positive was to remove the source buffer from the underlying model. If the model is only used by the query as I recall, I'm somewhat more inclined to move it to be located along with the ql file it is supporting rather than trying to make a general model that no one else is using. Thoughts?